src/app/connections/view/view.component.ts

```diff
@@ -43,11 +46,62 @@ export class ConnectionViewComponent implements OnInit, OnDestroy {
     private store: ConnectorStore,
     private formFactory: FormFactoryService,
     private formService: DynamicFormService,
+    private detector: ChangeDetectorRef,
   ) {
     this.loading = store.loading;
     this.connectors = store.list;
   }
 
+  showButton(id: string) {
+    switch (id) {
+      case 'salesforce':
+      case 'twitter':
+        return true;
+    }
+    return false;
+  }
+
+  doValidate(connector: Connector, formGroup: FormGroup) {
+    this.validateSuccess = false;
+    this.validateError = undefined;
+    this.validating = true;
+    const data = formGroup.value;
+    const sanitized: any = {};
+    // TODO for some reason some keys have spaces at the beginning
+    for ( const key in data ) {
+      if (!data.hasOwnProperty(key)) {
+        continue;
+      }
+      sanitized[key.trim()] = data[key] || '';
+    }
+    this.store.validate(connector.id, sanitized).subscribe((resp) => {
+      setTimeout(() => {
+        this.validating = false;
+        let errorHit = false;
+        (<Array<any>>resp).forEach((info) => {
+          if (!errorHit) {
+            if (info['status'] === 'ERROR') {
+              errorHit = true;
+              this.validateError = (<Array<any>>info)['errors'].map((err) => {
+                return err['description'];
+              }).join(', \n');
```

Comment:
Yeah, looks good ;-) For a next iteration we can be a more clever as for missing and invalid options we get the keys directly and so we could lookup the labels for the keys to create a nice and not so technical error message. 

This can be all done on the backend. So the structure of the error values will change to make it easier to consume. Something for the next sprint ;-)